src/lio/os/file.c

```diff
@@ -2516,9 +2545,20 @@ gop_op_status_t osfile_move_object_fn(void *arg, int id)
 
         log_printf(15, "ATTR sfname=%s dfname=%s\n", sfname, dfname);
 
-        err = rename(sfname, dfname);
+        if (lio_os_local_filetype(sfname) == 0)
+
+        err = rename(sfname, dfname);  //** Move the attribute directoy
+        if (err != 0) { //** Failed attr dir move so need to undo file entry
+           rename(dfname2, dfname);  //** restore the original dest file entry
+        } else {  //** Attr dir move was a success so go ahead and delete the original dest file entry
+           remove(dfname2);
+        }
     }
 
+fail:
+    osf_obj_unlock(lock_src);
+    if (lock_dest != NULL) osf_obj_unlock(lock_dest);
```

Comment:
Where's the style and naming guide?  I don't see one.  If there is one 
please point it out and we can discuss it.